mpire/comms.py

```diff
@@ -320,64 +418,38 @@ def get_results(self, block: bool = True, timeout: Optional[float] = None) -> An
         queue_empty_error = None
         with DelayedKeyboardInterrupt():
             try:
-                self._last_completed_task_worker_id, results = self._results_queue.get(block=block, timeout=timeout)
+                worker_id, results = self._results_queue.get(block=block, timeout=timeout)
                 self._results_queue.task_done()
+                if worker_id is not None:
+                    self._results_received[worker_id].value += 1
+                    self._last_completed_task_worker_id.append(worker_id)
             except queue.Empty as e:
                 queue_empty_error = e
```

Comment:
Ah. Much of the code isn't actually visible in the PR view, since it only shows difference and a few lines around it.

It still sounds a bit weird, because I'd imagine anything that's not KeyboardInterrupt would be unaffected by `DelayedKeyboardInterrupt`. But apparently it affects other exceptions as well?

Reply:
Well, it used to happen. I removed this part to verify, but it seems the queue.empty doesn't show up anymore on keyboard interrupt :/ Then I'll remove it now. Whenever it pops back up I know what to do ;)